File: actin-common/src/main/java/com/hartwig/actin/clinical/curation/config/ToxicityConfig.java
package com.hartwig.actin.clinical.curation.config;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class ToxicityConfig implements CurationConfig {

    @NotNull
    @Override
    public abstract String input();

    public abstract boolean ignore();

    @NotNull
    public abstract String name();

    @Nullable
    public abstract Integer grade();
}

File: clinical/src/main/kotlin/com/hartwig/actin/clinical/curation/config/ToxicityConfig.kt
package com.hartwig.actin.clinical.curation.config

import com.hartwig.actin.datamodel.clinical.IcdCode

data class ToxicityConfig(
    override val input: String,
    override val ignore: Boolean,
    val name: String,
    val grade: Int?,
    val icdCodes: Set<IcdCode>
) : CurationConfig